### PR TITLE
8342903: Deprecate for removal java.awt.Window.getWarningString()

### PR DESCRIPTION
**Pull request by prrace**
This fix deprecates for removal java.awt.Window.getWarningString() and also javax.swing.JInternalFrame.getWarningString()
l java.awt.Window.getWarningString() is only relevant with a SecurityManager and javax.swing.JInternalFrame.getWarningString() is just there for symmetry.

A few other spec changes ensue - the serial form variable warningString is removed is the main one.
The AWTPermission showWindowWithoutWarningBanner is obsolete
Reference to the system property  "awt.appletWarning" was actually already removed by JEP 486

There is a CSR for the spec. changes, which will need a reviewer : https://bugs.openjdk.org/browse/JDK-8344451

But the vast majority of the changes here are all the downstream obsoleted implementation
The warning string and warning icon, and warning windows are all obsoleted.
I think I tracked down all of it, but it is hard to be sure.
Automated tests are passing, and I've manually tested SwingSet2 on all 3 platforms.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Change requires CSR request [JDK-8344451](https://bugs.openjdk.org/browse/JDK-8344451) to be approved
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8342903](https://bugs.openjdk.org/browse/JDK-8342903): Deprecate for removal java.awt.Window.getWarningString() (**Bug** - P4)
 * [JDK-8344451](https://bugs.openjdk.org/browse/JDK-8344451): Deprecate for removal java.awt.Window.getWarningString() (**CSR**)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) Review applies to [cebd99ef](https://git.openjdk.org/jdk/pull/22221/files/cebd99efcc6494c6ef5d2b05a27164a354b8bfa4)
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**) Review applies to [425c2851](https://git.openjdk.org/jdk/pull/22221/files/425c2851fe4e3b3372b39b8f0be72a71af64cd17)
 * [Alexander Zuev](https://openjdk.org/census#kizune) (@azuev-java - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22221/head:pull/22221` \
`$ git checkout pull/22221`

Update a local copy of the PR: \
`$ git checkout pull/22221` \
`$ git pull https://git.openjdk.org/jdk.git pull/22221/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22221`

View PR using the GUI difftool: \
`$ git pr show -t 22221`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22221.diff">https://git.openjdk.org/jdk/pull/22221.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22221#issuecomment-2484251010)
</details>

### CODE REVIEW

File: make/modules/java.desktop/gensrc/GensrcIcons.gmk [hunk 1]
```diff
@@ -40,20 +40,6 @@ GENSRC_AWT_ICONS_SRC += \
 
 AWT_ICONPATH := $(MODULE_SRC)/share/classes/sun/awt/resources
```
Comment (erikj79) [inline make/modules/java.desktop/gensrc/GensrcIcons.gmk:41]: This variable looks to be unused now.

**Review by erikj79 — APPROVED** [review]
Hi @erikj79, thanks for making a comment in an OpenJDK project!

All comments and discussions in the OpenJDK Community must be made available under the OpenJDK [Terms of Use](https://openjdk.java.net/legal/tou/). If you already are an OpenJDK [Author](https://openjdk.java.net/bylaws#author), [Committer](https://openjdk.java.net/bylaws#committer) or [Reviewer](https://openjdk.java.net/bylaws#reviewer), please click [here](https://bugs.openjdk.java.net/secure/CreateIssue.jspa?pid=11300&issuetype=1) to open a new issue so that we can record that fact. Please Use "Add GitHub user erikj79" for the summary.

If you are not an OpenJDK Author, Committer or Reviewer, simply check the box below to accept the OpenJDK Terms of Use for your comments.

 - [ ]  I agree to the [OpenJDK Terms of Use](https://openjdk.java.net/legal/tou/) for all comments I make in a project in the [OpenJDK GitHub organization](https://github.com/openjdk).

Your comment will be automatically restored once you have accepted the OpenJDK [Terms of Use](https://openjdk.java.net/legal/tou/).

**Comment by azvegint** [comment]
I'll still be looking further, but it seems that there are few places that need more attention:

```shell
# grep for securitywarning
./open/src/jdk.compiler/share/data/symbols/java.desktop-8.sym.txt:7709:method name repositionSecurityWarning descriptor ()V flags 401
./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:206:    static boolean securityWarningEnabled;
./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:218:        initSecurityWarning();
./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:247:    static void initSecurityWarning() {
./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:252:        securityWarningEnabled = (runtime != null && runtime.contains("internal"));
./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:255:    static boolean isSecurityWarningEnabled() {
./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:256:        return securityWarningEnabled;
./src/java.desktop/unix/classes/sun/awt/X11/XBaseWindow.java:947:        if (XToolkit.isSecurityWarningEnabled() && XToolkit.isToolkitThread()) {
./src/java.desktop/windows/native/libawt/windows/awt.rc:32:// securityWarningIconCounter constant in awt_Toolkit.cpp.
```

`XToolkit` can also be cleaned up, and all usages of `XBaseWindow#checkSecurity()` can also be removed.

The `awt.rc` file looks like another candidate for removal, along with an update of `AwtLibraries.gmk`.

**Comment by prrace** [comment]
> I'll still be looking further, but it seems that there are few places that need more attention:
> 
> ```shell
> # grep for securitywarning
> ./open/src/jdk.compiler/share/data/symbols/java.desktop-8.sym.txt:7709:method name repositionSecurityWarning descriptor ()V flags 401
> ./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:206:    static boolean securityWarningEnabled;
> ./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:218:        initSecurityWarning();
> ./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:247:    static void initSecurityWarning() {
> ./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:252:        securityWarningEnabled = (runtime != null && runtime.contains("internal"));
> ./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:255:    static boolean isSecurityWarningEnabled() {
> ./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:256:        return securityWarningEnabled;
> ./src/java.desktop/unix/classes/sun/awt/X11/XBaseWindow.java:947:        if (XToolkit.isSecurityWarningEnabled() && XToolkit.isToolkitThread()) {
> ./src/java.desktop/windows/native/libawt/windows/awt.rc:32:// securityWarningIconCounter constant in awt_Toolkit.cpp.
> ```
> 
> `XToolkit` can also be cleaned up, and all usages of `XBaseWindow#checkSecurity()` can also be removed.
> 
> The `awt.rc` file looks like another candidate for removal, along with an update of `AwtLibraries.gmk`.



> I'll still be looking further, but it seems that there are few places that need more attention:
> 
> ```shell
> # grep for securitywarning
> ./open/src/jdk.compiler/share/data/symbols/java.desktop-8.sym.txt:7709:method name repositionSecurityWarning descriptor ()V flags 401
> ./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:206:    static boolean securityWarningEnabled;
> ./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:218:        initSecurityWarning();
> ./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:247:    static void initSecurityWarning() {
> ./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:252:        securityWarningEnabled = (runtime != null && runtime.contains("internal"));
> ./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:255:    static boolean isSecurityWarningEnabled() {
> ./src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:256:        return securityWarningEnabled;
> ./src/java.desktop/unix/classes/sun/awt/X11/XBaseWindow.java:947:        if (XToolkit.isSecurityWarningEnabled() && XToolkit.isToolkitThread()) {

I thought I'd got those (!) Must have not got back to them. So many trails to follow.
Fixing it just now I realise this is going to conflict with Harshita's PR : https://github.com/openjdk/jdk/pull/22218 - oh well .. I'll resolve that when I have to.

> ./src/java.desktop/windows/native/libawt/windows/awt.rc:32:// securityWarningIconCounter constant in awt_Toolkit.cpp.
> ```
> 
> `XToolkit` can also be cleaned up, and all usages of `XBaseWindow#checkSecurity()` can also be removed.

yes. checkSecurity can go.

> 
> The `awt.rc` file looks like another candidate for removal, along with an update of `AwtLibraries.gmk`.

It seems like it is just the refs to the SECURITY_WARNING icons (and the icon files themselves) that should be removed, so the file is edited but not removed and no change to the make file.

**Comment by prrace** [comment]
Updated per Alexander's comments

**Comment by prrace** [comment]
updated the EXCLUDE list as per Alexander's off line comment.

File: src/java.desktop/windows/native/libawt/windows/awt_Window.cpp [hunk 2]
```diff
@@ -854,195 +442,13 @@ bool AwtWindow::IsLayered(HWND window)
...
-                securityAnimationKind == akPreHide)) {
-        StartSecurityAnimation(akShow);
-    }
-    if (!show && ::IsWindowVisible(warningWindow)) {
-        StartSecurityAnimation(akPreHide);
-    }
+    return mrPassAlong;
 }
 
 void AwtWindow::FocusedWindowChanged(HWND from, HWND to)
```
Comment (azvegint) [inline src/java.desktop/windows/native/libawt/windows/awt_Window.cpp:450]: It looks like this empty method and its usage can also be removed:

```
./src/java.desktop/windows/native/libawt/windows/awt_Component.cpp:6426:    AwtWindow::FocusedWindowChanged(old, window);
```

Comment (prrace) [reply src/java.desktop/windows/native/libawt/windows/awt_Window.cpp:450]: I knew that was possible but I wasn't sure I wanted to pull on the string.
If it is really simple I will do it.

**Comment by prrace** [comment]
> There are a few minor mentions of the Warning Window left
> 
> ```
> ./src/java.desktop/unix/classes/sun/awt/X11/XWindow.java:1492:     * XWarningWindow, both of which have the XWindow as a common ancestor.

I've updated the comment
> ./test/jdk/ProblemList.txt:468:java/awt/security/WarningWindowDisposeTest/WarningWindowDisposeTest.java 8266059 macosx-all
> ```
> 
> the test in problem list does not exist

I am fairly sure this test was deleted by the JEP 486 integration. It should have been removed then.

I'll fix it and close out the bug

File: src/java.desktop/windows/native/libawt/windows/awt_Window.cpp (hunk 2 above)
Comment (prrace) [reply src/java.desktop/windows/native/libawt/windows/awt_Window.cpp]: > I knew that was possible but I wasn't sure I wanted to pull on the string. If it is really simple I will do it.

OK. The string was very short. Fixed.
I believe all comments are now addressed.

**Review by azuev-java — APPROVED** [review]
Latest revision looks good.

**Review by azvegint — APPROVED** [review]

**Pull request closed without merging**